app/workers/soft_block_worker.rb

```diff
@@ -0,0 +1,17 @@
+# frozen_string_literal: true
+
+class SoftBlockWorker
+  include Sidekiq::Worker
+
+  sidekiq_options queue: 'pull'
+
+  def perform(account_id, target_account_id)
+    account        = Account.find(account_id)
+    target_account = Account.find(target_account_id)
+
+    BlockService.new.call(account, target_account)
```

Comment:
The block followed by unblock here has the effect of:

- Remove any follows between the accounts, in both directions
- Create a block from account to target
- Clear the timeline and notifications of the account from anything from the target
- Salmon the block from the account to the target
- ...
- Remove the block
- Send the unblock salmon

This does not prevent accounts on the target domains from immediately refollowing the person, unless they also change their account to private, right?

I just want to make sure I understand the general flow here.